src/dask_awkward/lib/core.py

```diff
@@ -921,6 +921,36 @@ def __dask_postpersist__(self):
 
     __dask_scheduler__ = staticmethod(threaded_get)
 
+    # taken from https://github.com/dask/dask/blob/3003db5b84070b1afc197c7c70c76c5c4c2bc821/dask/array/core.py#L1854-L1883
+    def __bool__(self):
+        materialized = self.compute().to_numpy()
+        if materialized.size != 1:
```

Comment:
Can we know the `size` before compute? It seems awfully wasteful to run a graph only then to realise that we can't use the output.

Reply:
Hm. I am sorely tempted to say we should `raise NoteImplemented` for _all_ these methods ...

Reply:
I agree. I can add a `TracerConversionError` (similar to what JAX does) and raise it with a helpful message for all of these methods. I'll update the PR accordingly if you're fine with this procedure.

(Then dask-awkward and dask behave differently though, which might be confusing for people coming from dask. Do you think this is a problem?)

Reply:
I am not sure what the type of the exception should be, so long as the message it delivers is clear.